app/services/file_path_sanitizer.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class FilePathSanitizer
+  def sanitize(file_path, on_error: nil)
+    pathname = Pathname.new(file_path)
+    return pathname.realpath if  pathname.file?
+
+    on_error&.call
+    false
+  end
+end
```

Comment:
Okay, that's something. I thought that `validate_file_path` would go here, too. But I don't want to hinder this PR. It's fixing a security issue. :+1: